dora/src/semck/typeparamck.rs

```diff
@@ -118,7 +118,7 @@ impl<'a, 'ast> TypeParamCheck<'a, 'ast> {
         let cls = cls.read();
 
         for &trait_bound in &tp.trait_bounds {
-            if !cls.traits.contains(&trait_bound) {
+            if !cls.implements_trait(self.vm, trait_bound) {
```

Comment:
This works for classes but not for tuples. Tuples never implement Zero, but we can also solve this in the next PR.